src/main/python/vantiqsdk.py

```diff
@@ -0,0 +1,1399 @@
+
+    As noted, most of the operations in the Vantiq object are `async` and must be awaited for them to run. See
+    https://docs.python.org/3/library/asyncio.html for more details about working with async operations.
+
+    """
+
+    def __init__(self, server: str, api_version: Union[str, None] = None):
+        """Create a Vantiq client object.
+
+        Parameters:
+            server : str
+                URL String at which to find the Vantiq Server
+
+            api_version : str (optional)
+                Version of the API to uwe. Defaults to '1'
+
+        Returns:
+            Vantiq client object with which to interact with the Vantiq system.
+
+        Can be used as a context manager
+
+        Examples:
+            As a context manager::
+            ::
+
+                async with Vantiq('https://dev.vantiq.com') as client:
+                    ...
+
+            or, in a client object:
+            ::
+                client = Vantiq('https://dev.vantiq.com')
+                ...
+                await client.close()
+
+
+        """
+        if exists('logger.ini'):
+            logging.config.fileConfig('logger.ini', disable_existing_loggers=False)
+        else:
+            print('No logger.ini file found.')
+        self._vlog: Logger = logging.getLogger(self.__class__.__name__)
+
+        if server.endswith('/'):
+            self._server = server[:-len('/')]  # server.removesuffix('/')'
+        else:
+            self._server = server
+        self._api_version = api_version if api_version is not None else '1'
+        self._is_authenticated = False
+        self._username = None
+        self._access_token = None
+        self._id_token = None
+        self._auth_header = None
+        self._connection = None
+        self._is_connected = False
+        self._connection = _RestClient(self._server)
+        self._base_path = '/api/v' + self._api_version + '/'
+        self._subscriber: Union[_VantiqSubscriber, None] = None
+
+    async def __aenter__(self):
+        await self.connect()
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.close()
+
+    async def connect(self) -> None:
+        """Make the connection to the Vantiq server
+
+        Returns:
+            None
+        Raises:
+            VantiqException for connection issues
+
+        Notes:
+            Should precede the .authenticate().  Must be done before other operations.
+        """
+
+        if not self._is_connected:
+            await self._connection.connect()
+            self._is_connected = True
+            self._vlog.debug('Connected to server: %s', self._server)
+
+    def get_server(self):
+        """Returns the server url used for this connection."""
+        return self._server
+
+    def get_api_version(self):
+        """Returns the API version used for this connection."""
+        return self._api_version
+
+    async def set_access_token(self, access_token: str) -> None:
```

Comment:
Does the use of `await` make this async?  From previous context I thought that `await` forced a synchronous wait.  Does that become async when the caller doesn't use `await`?

Reply:
Nvm -- RTFM and I think I'm good now

Reply:
Yeah, it's tricky to read.  But sort of like setting up an rxchain -- `await` is the subscribing.  But you can also collect them up then do an asycio.gather(*list of awaitables) to wait for any/all of them.